tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc

```diff
@@ -6424,168 +6425,209 @@ auto get_split_nodedef = [](DataType dtype, int num_split) -> NodeDef {
-      {/*input_shape=*/{1, 6},
-       /*value=*/common_input,
-       /*axis=*/2,
-       /*num_split=*/6,
-       /*expected_output_dims=*/{1, 1},
-       /*expected_outputs=*/
-       {{CType(0)},
-        {CType(1)},
-        {CType(2)},
-        {CType(3)},
-        {CType(4)},
-        {CType(5)}}},
-      {/*input_shape=*/{1, 6},
-       /*value=*/common_input,
-       /*axis=*/-1,
-       /*num_split=*/2,
-       /*expected_output_dims=*/{1, 3},
-       /*expected_outputs=*/
-       {InitTestVector<CType>(3), InitTestVector<CType>(3, CType(3))}},
-  };
+  // Create inputs.
+  test->AddTestWeights("axis", {1}, {p.axis}, DT_INT32);
+  test->AddTestTensor("value", p.input_shape, test->get_tf_type(),
+                      p.input_value);
 
-  for (int i = 0; i < ok_params.size(); ++i) {
-    test->Reset();
-    NodeDef node_def = get_split_nodedef(dtype, ok_params[i].num_split);
-    // Create inputs.
-    test->AddTestWeights<int32>("axis", {1}, {ok_params[i].axis});
-    nvinfer1::DataType trt_type;
-    TF_ASSERT_OK(TfTypeToTrtType(dtype, &trt_type));
-    test->AddTestTensor("value", ok_params[i].input_shape, 1, trt_type);
-    // Convert.
-    test->RunValidationAndConversion(node_def);
+  std::vector<Matcher<std::vector<float>>> matcher_vec;
+  std::vector<DataType> datatype_vec;
+  std::vector<std::vector<int>> expected_output_dims;
 
-    // Get output tensors and verify output dims.
-    EXPECT_EQ(ok_params[i].expected_outputs.size(), ok_params[i].num_split);
-    std::vector<TRT_TensorOrWeights> outputs(ok_params[i].num_split);
-    DataVec output_data;
-    for (int j = 0; j < outputs.size(); ++j) {
-      const string name = j == 0 ? StrCat("my_split") : StrCat("my_split:", j);
-      TF_EXPECT_OK(test->GetTensorOrWeights(name, &outputs[j]));
-      EXPECT_TRUE(outputs[j].is_tensor());
-      ExpectTrtDimsEqualsArray(ok_params[i].expected_output_dims,
-                               outputs[j].tensor()->getDimensions());
-      // Create buffer to store output.
-      output_data.push_back(
-          {name, test->ConstructTensor<CType>(
-                     ok_params[i].expected_outputs[j].size())});
-    }
+  for (int j = 0; j < p.expected_outputs.size(); ++j) {
+    matcher_vec.push_back(ElementsAreArray(p.expected_outputs[j]));
+    datatype_vec.push_back(test->get_tf_type());
+    expected_output_dims.push_back(p.expected_output_dims);
+  }
 
-    // Verify output values are correct.
-    const DataVec input_data{
-        {"value", test->AsTensor<CType>(ok_params[i].value)}};
-    TF_EXPECT_OK(test->BuildAndRun(input_data, &output_data));
-    for (int j = 0; j < outputs.size(); ++j) {
-      EXPECT_THAT(GetSpanForData<CType>(output_data[j]),
-                  ElementsAreArray(ok_params[i].expected_outputs[j]));
-    }
+  Status run_status;
+  if (p.error_code == error::OK) {
+    run_status = Status::OK();
+  }
+  else {
+    auto expected_msg_substr = absl::StrFormat(p.expected_msg_substr, p.axis);
```

Comment:
It turns out that his syntax doesn't work because within google, absl::StrFormat requires the string to be constexpr.

Reply:
Fixed